docs/index.md

```diff
@@ -79,26 +77,26 @@ def test_hello_world():
         assert result == f"Hello, {random_input}"
 ```
 
-Let's analyze the `test_hello_world()` function:
+The `test_hello_world()` function demonstrates key aspects of testing with `algopy-testing-python`, following the Arrange-Act-Assert pattern:
+
+1. Arrange:
+
+    - Simulated Environment: Utilizes `algopy_testing_context()` to mimic the Algorand Virtual Machine. It also provides access to an array of utilities and so called `anonymous variables` which allow quick instantiation of random values of specific types where the value itself is not important, it also serves as a backbone for property based testing capabilities.
+    - Test Data Generation: Leverages `ctx.any_string()` for creating random, typed inputs of type `algopy.String`. Note that this differs from `algopy.arc4` types, for which context manager provides separate `any_*` methods accessible via `arc4` property on context instance.
+    - Contract Instantiation: Creates an instance of `HelloWorld` contract within the test context. Test context automatically instantiates a corresponding `algopy.Application` and links it with the class instance which is loaded in the python interpreter.
 
-1. Function Definition: Tests the `HelloWorld` contract.
-2. Testing Context: Uses `algopy_testing_context()` to simulate an Algorand environment.
-3. Test Setup:
-    - Generates a random 10-character string input of type `algopy.String`.
-    - Instantiates the `HelloWorld` contract.
-4. Contract Interaction:
-   Invokes the contract's `hello()` method with the random input. This occurs within a Python interpreter, not on the actual Algorand network. This provides the ability to inspect the contract's state and debug execution as a regular Python program.
-5. Assertion:
-   Verifies the expected behavior: `assert result == f"Hello, {random_input}"`.
+2. Act:
 
-Key features of `algopy_testing_context`:
+    - Method Invocation: Calls `hello()` in a Python interpreter, enabling state inspection and debugging. When you run methods decorated with `abimethod` or `baremethod`, test context will automatically assemble a respective `appl` transaction which can be accessed via `context.get_active_transaction`, note that it will also automatically pre populate foreign references passed, as well as related method signature simulating AVM behaviour.
```

Comment:
```suggestion
    - Method Invocation: Calls `hello()` in a Python interpreter, enabling state inspection and debugging. When you run methods decorated with `abimethod` or `baremethod`, test context will automatically assemble a respective `appl` transaction which can be accessed via `context.get_active_transaction`, note that it will also automatically pre populate foreign references passed, as include the appropriate method signature in the application args.
```